[Blog] Introducing OSCAR: The AI Chat-bot that Enables OpenSearch Conversational Automation for Releases (#3929)
opensearch-project/project-website: _posts/2025-09-03-Introducing-OSCAR:-The-AI-Chat-bot-that-Enables-OpenSearch-Conversational-Automation-for-Releases.md

---
layout: post

> title: "Introducing OSCAR: The AI Chat-bot that Enables OpenSearch Conversational Automation for Releases"

Reviewer: ```suggestion
title: "Introducing OSCAR: The AI chatbot enabling conversational automation for OpenSearch releases"
```

category: blog
tags: [opensearch-3-2]
authors:
    - divsen
    - zhujiaxiang
date: 2025-09-03
categories:
  - technical-posts
meta_keywords: AI Chatbot, Conversational AI, Release Automation, OpenSearch Community, Jenkins Integration, LLM, AI Agents, OpenSearch Engineering, Software Release Management, Open-Source Project, Secure AI, Linux Foundation, OpenSearch 3.2
meta_description: Discover OSCAR, the AI chatbot designed to automate and democratize the OpenSearch release process. Learn how this multi-agent system uses AI to provide intelligent assistance, automate Jenkins workflows, and simplify release management for developers and community contributors.
---

> OpenSearch is an open-source search and analytics suite, used as a platform for website search, log analytics, and application monitoring. It is powered by the OpenSearch core engine, OpenSearch Dashboards for visualization, and a suite of plugin applications that solve different use cases.

Reviewer: ```suggestion
OpenSearch is an open-source search and analytics suite used as a platform for website search, log analytics, and application monitoring. It is powered by the OpenSearch core engine, OpenSearch Dashboards for visualization, and a suite of plugin applications that solve different use cases.
```

> The OpenSearch project is a community-driven initiative, announced in January 2021 and adopted by the Linux Foundation since September 2024. This initiative oversees the open governance, collaboration, innovation, and development for the entire OpenSearch suite, including OpenSearch core and OpenSearch Dashboards.

Reviewer 2: The OpenSearch project is community-driven. Since the project was adopted by The Linux Foundation in Sept 2024, repositories have expanded from 120+ to over 170, active contributors have risen from 1000+ to over 3,500, and today, total downloads have exceeded 1.1 billion.

With exponential growth in community contributions, particularly since joining the Linux Foundation, the OpenSearch project has also evolved its build and release infrastructure. This growth includes expanding from 120+ to over **170 repositories** and a surge in active contributors from 1,000+ to over **3,500**.

> This expansion, which has pushed total downloads beyond **1.1 billion**, has created a sophisticated system with multiple components, extensive testing requirements, and complex release orchestration. The OpenSearch Engineering Effectiveness (OSEE) team is responsible for driving OpenSearch releases in collaboration with the community while resolving engineering issues, bottlenecks, and inefficiencies across the project.

Reviewer 2: With momentum comes increased complexities. Release issues need to ship on time, and involve multiple components, extensive testing, and careful orchestration every eight weeks.

The OpenSearch Engineering Effectiveness (OSEE) team helps the community run releases, resolves engineering issues, and removes bottlenecks and inefficiencies across the project.

### Architectural overview

OSCAR's architecture, shown in the following image, implements a dual-agent security model, in which privileged users access full system capabilities while standard users receive read-only assistance, ensuring both accessibility and security.

![OSCAR Architecture Image](/assets/media/blog-images/2025-09-03-Introducing-OSCAR:-The-AI-Chat-bot-that-Enables-OpenSearch-Conversational-Automation-for-Releases/oscar_architecture.png)

### Component flow for a release workflow: Jenkins query

OSCAR employs communication between multiple components, including multiple agents, dynamo DB, and knowledge bases, shown in the following image.

![Jenkins Workflow Image](/assets/media/blog-images/2025-09-03-Introducing-OSCAR:-The-AI-Chat-bot-that-Enables-OpenSearch-Conversational-Automation-for-Releases/jenkins_query_flow.png)

## OSCAR in action: 3.2.0 release success stories

> During the OpenSearch 3.2.0 release, we launched a beta version of OSCAR internally. OSCAR demonstrated its value across three critical areas:

Reviewer 2: During the 3.2 release, we utilized a beta version of OSCAR to drive routine steps with natural language.

**Modular infrastructure**: Using AWS CDK with separate stacks enabled rapid iteration on individual components. The serverless Lambda architecture automatically handled scaling during release periods, while DynamoDB's TTL features managed conversation context without manual cleanup.

**Real-world complexity**: The 3.2.0 release revealed gaps between assumptions and actual needs. Users required intelligent deduplication, automatic parameter resolution, and context-aware formatting, driving sophisticated query routing and specialized response logic.

**Key insight**: Successful AI automation for enterprise workflows requires not just intelligent responses, but intelligent orchestration of multiple systems with robust security and deep understanding of operational complexity.

## Beta status and public availability

OSCAR is currently in beta, actively supporting OpenSearch release processes while the development team refines features and gathers user feedback. The system has proven its value during the 3.2.0 release cycle, demonstrating significant improvements in efficiency and accessibility.

> ### How Can I Use OSCAR?

Reviewer: ```suggestion
### How can I use OSCAR?
```

* **Threat intelligence integration**: Proactive security notifications using real-time threat feeds.

### Dynamic knowledge management

These improvements aim to centralize and intelligently surface OpenSearch knowledge, making information accessible and actionable for contributors and maintainers:

* **Comprehensive documentation ingestion**: Automated ingestion from all OpenSearch repositories, release notes, and blog posts with semantic understanding.
* **Version-aware knowledge**: Intelligent routing of documentation based on user context and version requirements.
* **Self-healing infrastructure**: Autonomous issue detection and resolution with pattern recognition and predictive maintenance.

### Upcoming milestones

These are the next steps in rolling out OSCAR to the community, expanding access, and improving adoption:

* **Public Slack integration**: Making OSCAR available in public OpenSearch Slack channels.
* **Community access**: Expanded access for external contributors and community release managers.
* **Enhanced documentation**: Comprehensive guides for community adoption and contribution.

> We're committed to maintaining OSCAR's pioneering approach while ensuring stability and security as we expand access to the broader OpenSearch community. If you are interested in learning more about OSCAR, join the [OpenSearch Slack Workspace](https://opensearch.org/slack/) and try it out for yourself!

Reviewer 2: **Call to action**

If you contribute to OpenSearch, try a release task with OSCAR as your copilot. If you are new to releases, pair with a mentor for your first step. If you run releases in another community, bring your best ideas to our RFCs. Together we can keep the cadence, raise the quality bar, and make participation easier for everyone.
